File: camouflage.py
from collections import defaultdict

def solution(clothes):
    d = defaultdict(list)
    key = []
    for i in range(len(clothes)):
        d[clothes[i][1]].append(clothes[i][0])
        if clothes[i][1] not in key:
            key.append(clothes[i][1])
    print(key)
    print(len(d))
    answer = 0
    
    return answer

File: camouflage.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dic : [String : Int] = [:]
    for clothe in clothes {
        if dic[clothe[1]] != nil {
            dic[clothe[1]]! += 1
        } else {
            dic[clothe[1]] = 1
        }
    }
    var answer = 1
    dic.keys.forEach{ answer *= (dic[$0]! + 1) }
    return answer - 1
}